care/utils/assetintegration/onvif.py

```diff
@@ -1,13 +1,149 @@
 import enum
+import json
 
-from rest_framework.exceptions import ValidationError
+from django.core.cache import cache
+from rest_framework.exceptions import PermissionDenied, ValidationError
 
+from care.users.models import User
 from care.utils.assetintegration.base import BaseAssetIntegration
 
 
 class OnvifAsset(BaseAssetIntegration):
     _name = "onvif"
 
+    class UsageManager:
+        def __init__(self, asset_id: str, user: User):
+            self.asset = str(asset_id)
+            self.user = user
+            self.waiting_list_cache_key = f"onvif_waiting_list_{asset_id}"
+            self.current_user_cache_key = f"onvif_current_user_{asset_id}"
+
+        def get_waiting_list(self) -> list[User]:
+            asset_queue = cache.get(self.waiting_list_cache_key)
+
+            if asset_queue is None:
+                return []
+
+            return [
+                user
+                for username in asset_queue
+                if (user := User.objects.filter(username=username).first()) is not None
```

Comment:
why not use `.exists` and `.get` instead

Reply:
use .filter(username__in=asset_queue)